src/test/ui/consts/control-flow/basics.rs

```diff
@@ -0,0 +1,43 @@
+// Test basic functionality of `if` and `match` in a const context.
+
+// run-pass
```

Comment:
More test ideas (for other files -- and in other follow up PRs if necessary):

- Test all the various https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/enum.PatKind.html forms and some random walk of that. (No reason to believe it's different, but sometimes that's wrong; well except `Box` patterns).

- Also test `if let` (good idea to not assume a HIR->MIR desugar since that might change, and probably will...).

Reply:
@jyn514, this would good be a good place to contribute if you're still motivated. "Interesting" cases could be added to `basics.rs` (or another file) that use more types of patterns in `match` and `if let`. If you push to a branch I can cherry pick them into this PR (keeping you as the author of course), or you can open a separate PR once this is merged.